language/move-lang/tests/move_check/parser/let_binding.move

```diff
@@ -0,0 +1,29 @@
+module M {
+    struct R {
+        f: u64
+    }
+    struct Generic<T> {
+        g: T
+    }
+    f() {
+        let () = ();
+        let (): () = ();
+        // FIXME: The grammar currently treats "()" as a single token, but it is
```

Comment:
The only one is I probably don't want spaces between the type for `unit`, i.e. it should always be `foo(): ()` not `foo(): (   )`

Reply:
I disagree on this one. The new parser treats "(" and ")" as separate tokens. It's just simpler and more consistent that way. We could add a special check of the source locations to detect whitespace between them, but I don't even think that's a good idea. There may be some cases, e.g., code generated from a macro preprocessor where there are types conditionally inserted between the parenthesis, where forbidding whitespace here would be a hassle. I wouldn't ever recommend writing the unit type with whitespace, but it doesn't seem like we need to actively disallow it.